packages/web-app-files/src/views/trash/Overview.vue

```diff
@@ -0,0 +1,244 @@
+  <div v-else class="oc-flex">
+    <files-view-wrapper>
+      <app-bar
+        :breadcrumbs="breadcrumbs"
+        :has-sidebar-toggle="false"
+        :has-view-options="false"
+        :has-hidden-files="false"
+        :has-file-extensions="false"
+        :has-pagination="false"
+      />
+      <div>
+        <oc-text-input
+          id="spaces-filter"
+          v-model="filterTerm"
+          class="oc-ml-m oc-my-s"
+          :label="$gettext('Search')"
+          autocomplete="off"
+        />
+        <oc-table
+          ref="tableRef"
+          class="spaces-table"
+          :sort-by="sortBy"
+          :sort-dir="sortDir"
+          :fields="fields"
+          :data="orderedSpaces"
+          :sticky="true"
+          :hover="true"
+          @sort="handleSort"
+        >
+          <template #icon="{ item }">
+            <oc-icon class="oc-pl-m" v-if="item.driveType === 'personal'" name="folder" />
+            <oc-icon class="oc-pl-m" v-else name="layout-grid" />
+          </template>
+          <template #name="{ item }">
+            <oc-button
+              class="oc-display-block"
+              type="router-link"
+              appearance="raw"
+              :to="getTrashLink(item)"
+              v-text="getSpaceName(item)"
+            />
+          </template>
+          <template #footer>
+            <div class="oc-text-nowrap oc-text-center oc-width-1-1 oc-my-s">
+              <p class="oc-text-muted">{{ footerTextTotal }}</p>
+              <p v-if="filterTerm" class="oc-text-muted">{{ footerTextFilter }}</p>
+            </div>
+          </template>
+        </oc-table>
+      </div>
+    </files-view-wrapper>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, nextTick, onMounted, ref, unref, watch } from 'vue'
+import { buildSpace } from 'web-client/src/helpers'
+import Mark from 'mark.js'
+import Fuse from 'fuse.js'
+import { useGettext } from 'vue3-gettext'
+import { useTask } from 'vue-concurrency'
+import {
+  configurationManager,
+  useCapabilityProjectSpacesEnabled,
+  useGraphClient,
+  useRouter
+} from 'web-pkg'
+import { createLocationTrash } from 'web-app-files/src/router'
+import { createFileRouteOptions } from 'web-pkg/src/helpers/router'
+import AppBar from 'web-app-files/src/components/AppBar/AppBar.vue'
+import FilesViewWrapper from 'web-app-files/src/components/FilesViewWrapper.vue'
+
+export default defineComponent({
+  name: 'TrashOverview',
+  components: { FilesViewWrapper, AppBar },
+  methods: { createLocationTrash },
+  setup: function () {
+    const router = useRouter()
+    const { $gettext } = useGettext()
+    const sortBy = ref('name')
+    const sortDir = ref('asc')
+    const filterTerm = ref('')
+    const markInstance = ref(undefined)
+    const tableRef = ref(undefined)
+    const spaces = ref([])
+    const { graphClient } = useGraphClient()
+    const hasProjectSpaces = useCapabilityProjectSpacesEnabled()
```

Comment:
Please introduce a personal space computed prop below this line:
```js
const personalSpace = computed(() => {
      return store.getters['runtime/spaces/spaces'].find((s) => isPersonalSpaceResource(s))
    })
```
(needed down below)